### ISSUE TITLE
JSON field not compatible with built-in query batching

### ISSUE THREAD

**Issue by kpsuperplane** [prisma/prisma#2815]
## Bug description

#1470 introduced batching/dataloader pattern for `findOne`. It appears this is not compatible with `Json` fields for some reason (specifically the nullable `Json?` field)

I can confirm this is most likely the case because the exact same code succeeds when there's only one `findOne` call.  

## How to reproduce

Consider table `A` with a column of `Json?` type. 

Imagine we have the following code:
```
const ids = [1, 2, ...];
const items = await Promise.all(ids.map(id => prisma.A.findOne({where: { id }})));
```

If the ids array has length 1, it will succeed. 
If the ids array has length > 1, it will fail with the following error: 
```
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at IncomingMessage.<anonymous> (engine-core/dist/h1client.js:48:1)
    at IncomingMessage.emit (events.js:322:22)
    at IncomingMessage.EventEmitter.emit (domain.js:482:12)
    at endReadableNT (_stream_readable.js:1187:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

**Note** At this point I have to restart the entire server, subsequent queries of any kind continue to fail. 

## Expected behaviour

It should not crash. 

## Prisma information

Notably with `log: ["query"]` it doesn't even have a chance to output the query before crashing. 

## Environment & setup

OS: Latest Mac OS
Database: Postgres 9.6
Prisma Version: 2.1.0-dev.52
Node.JS Version: 12.16.3

**Comment by pantharshit00** [comment]
Hi,

Thanks for the report. I actually hit this error while I was triaging some other issue but I wasn't able to isolate it. This reports helps a lot. 

So there is a regression in the dev channel. If I run the batch in version `2.0.1` it works fine. We will need to look into this.

Reproduction for internal use: https://github.com/harshit-test-org/prisma-issue-2815

So please roll back to 2.0.1 while we fix this.

**Comment by timsuchanek** [comment]
Thanks for reporting!
As fast as this bug came, it also left us.
I just tried your reproduction @pantharshit00, which I can reproduce for the version you provided - `2.1.0-dev.53`. Upgrading it to the latest dev, `2.0.0-dev.55` makes is succeed.

**Comment by jgkingston** [comment]
Hi,

We are still encountering this issue when using `findOne` in `_resolveReference` with Apollo Federation. We have tried with version `2.0.1` as suggested above and with `2.5.1`. We have also tried with both nullable and non-nullable `Json` fields.

Replacing `findOne` with `findMany` and returning the first item does work however as does a single request that resolve with `findOne` so the issue seems to be specific to batched requests using `findOne`. 

Thanks
